BDArmory/Weapons/Missiles/MissileLauncher.cs

```diff
@@ -1349,7 +1334,7 @@ public override void FireMissile()
             {
                 SourceVessel = vessel;
             }
-            var wpm = VesselModuleRegistry.GetMissileFire(SourceVessel, true);
+            var wpm = SourceVessel.ActiveController().WM;
             if (wpm != null) Team = wpm.Team;
```

Comment:
The WMs in FireMissile() would want to be the WM at time of firing, not whatever the current active WM is, to ensure proper targeting, so would warrant passing along the WM from missileFire via FireMissile(missileFire WM) instead of assigning it here. Admittedly something of an edge case as it would require a WM change at the exact same time as a launch

Reply:
Can that change happen? All the `MissileLauncher.FireMissile()` calls are made immediately after `SendTargetDataToMissile` calls without any `yield`s, so there shouldn't be any event that switches the active WM while firing the missile.